### ISSUE TITLE
[Bug] Scrolling doesn't work after pausing

### ISSUE THREAD

**Issue by garrett** [mozilla-mobile/fenix#14527]
## Steps to reproduce

1. Attempt to scroll a page.
2. Sometimes, ~when the URL bar is visible,~ scrolling doesn't work. _This is usually when you stop scrolling for a bit and try scrolling again._
3. A second attempt to scroll does work, however.

### Expected behavior

Scrolling, as Firefox did before a few days ago (relative to the opening date of this issue).

### Actual behavior

Not registering swipes as scrolls ~when the URL bar is fully visible~.

### Device information

* Android device: OnePlus 6T
* Fenix version: 200831 16:01 (Build #2015761139)

This has been an issue for roughly a week (or so)? It worked fine before. I think this happened with a recent change with the implementation of the URL bar hiding _or_ with the new tab screenshot generation. (Perhaps it causes a pause and doesn't register scrolling?)

**Comment by garrett** [comment]
I quickly skimmed through the recent commits over the past few days and found something that _might_ be causing this:
https://github.com/mozilla-mobile/fenix/commit/73321eb2bb4fbb79abb294fe5853d2b34e5f34f1

It's supposed to hide the keyboard but has a touch listener. Could it be getting in the way? :shrug:

My timeframe could be off, as I was on vacation for the past week. This might've been a problem for a few days longer. But I'm pretty sure the issue did show up in builds over the past week.

**Comment by TheLounger** [comment]
Can confirm. When address bar is visible, scrolling (often?) doesn't work on the first touch.

This also happens often when the address bar is hidden and you want to scroll. Instead it shows the address bar and you have to touch/scroll *again* for it to work. It also seems to do this when I want to click a link.

**Comment by garrett** [comment]
As Nightly has the option to not auto-hide the URL bar, I enabled that and tried scrolling pages. It still happens, even in 
reader mode. So this probably has nothing to do with hiding or showing the URL bar. (But it may still be the result of 73321eb.)

I'm actually extremely surprised that this hasn't been fixed yet... Scrolling is a rather major activity everyone does on every website. At least for me, this was a dealbreaker to run Nightly (which I have been running for the majority of a year until now), so I had to switch back to Beta. Hopefully it gets fixed soon so it doesn't leak into Beta... and then into Stable.

**Comment by garrett** [comment]
I've updated the title to reflect the fact that it doesn't seem to be URLbar related.

I hit this issue when I scroll a bit, then stop scrolling for a bit, then start scrolling again.

**Comment by garrett** [comment]
I've been trying to figure out exactly what is happening, trying to recreate the conditions causing this bug from happening. It does happen in Nightly, it does _not_ happen in Beta (yet), and doesn't happen in Release.

I haven't been able to recreate it with the back, home, and next navigation (yet), but have experienced it a ton with gesture navigation. However, sometimes it just doesn't show up and other times it happens continually.

Basically, it happens like this:

1. try to scroll a page... nope! it doesn't work
2. try to scroll a page _again_... it works!
3. wait a few seconds (for inertia to stop)
4. repeat (it doesn't work the first time, then it works after that)

When the URL bar is hidden, it brings up the URL bar, but doesn't scroll the page. When the URL bar is constantly shown, this not-scrolling swipe still happens, however.

**Comment by garrett** [comment]
OK, so I realized I had turned on WebRender with the override on Nightly, and I don't have it turned on with Beta. I'll try browsing a while with it off and see if I hit this again. (If so... :man_facepalming: @ self.)

**Comment by eliserichards** [comment]
Does this happen on every web page? (do you have an example of a site that consistently repros?)

**Comment by garrett** [comment]
Yep, this can happen on every page.

I've mainly tested it on Pocket articles, because it's a prominent default icon in Firefox (and I use Pocket too). But I've seen it on other sites as well — usually news sites, as those require a lot of scrolling.

It happens in Firefox's reader mode as well.

I just reset all my storage via Android's app settings for Firefox... and I still hit the bug.

**Comment by kbrosnan** [comment]
> I quickly skimmed through the recent commits over the past few days and found something that _might_ be causing this:
> [73321eb](https://github.com/mozilla-mobile/fenix/commit/73321eb2bb4fbb79abb294fe5853d2b34e5f34f1)
> 
> It's supposed to hide the keyboard but has a touch listener. Could it be getting in the way? 🤷
> 
> My timeframe could be off, as I was on vacation for the past week. This might've been a problem for a few days longer. But I'm pretty sure the issue did show up in builds over the past week.

That change was in the search/address bar UI. This sounds like Gecko web content interactions.

**Comment by pocmo** [comment]
> That change was in the search/address bar UI. This sounds like Gecko web content interactions.

Either that or maybe our gesture detector, @Mugurell?

**Comment by snorp** [comment]
I've never seen this behavior on Pixel 4 XL. Garrett, what device are you using? If you have other devices, does it reproduce there as well?

**Comment by staktrace** [comment]
Probably worth trying to see if it happens in the GVE app, and if so, bisect there using mozregression. If it doesn't happen in GVE it's likely to be on the Fenix side rather than Gecko-side web content interaction.

**Comment by garrett** [comment]
This is on a OnePlus 6T, and I have gesture navigation on.

Since clearing the storage for Firefox, I did notice touches not being registered when scrolling, while I was syncing via Firefox Sync. Now that the sync seems to have finished, I haven't been able to recreate this scrolling issue.

A guess: Do you think Firefox might have been dropping touches at times when it was hitting IO pretty hard? And my data (which was a lot, as I had been using Nightly for quite some time) included a heavily fragmented or corrupt sqlite DB that made it more pronounced?

Basically, a mobile version of https://support.mozilla.org/en-US/kb/firefox-hangs-or-not-responding

If this is the case, I guess having the Firefox refresh feature on mobile might not be a bad idea.

Meanwhile, I set my default browser back to Firefox Nightly and will see if it happens again.

I'm sorry for making noise. It was pretty severe — but I guess a corner case, not a regression? Especially since no one here can reproduce it. (And now scrolling is apparently fine again in Nightly for me too.)

 :heart: Thanks, everyone, for working on Firefox. :heart:

**Comment by garrett** [comment]
Quick follow-up: It indeed is still happening... perhaps not quite a frequently as before for me.

I just hit it again after scrolling on the https://support.mozilla.org/en-US/kb/firefox-hangs-or-not-responding page on my phone.

**Comment by eliserichards** [comment]
Thanks for all of the info @garrett!

**Comment by garrett** [comment]
Further testing: 
- I do see the no-scroll problem with the three button navigation (non-gesture) as well. So it's not related to gesture support.
- It does feel like it happens less frequently after clearing my data, but it's still here (after syncing — so admittedly, I could possibly have some issue in my data that's sync'd back over).
- After the first time it happens, it seems like it happens more frequently for a while.

I have some older Android phones where I will install and test Nightly as well.

**Comment by peter-vasut** [comment]
Not sure if this helps, but I can confirm same issue on LG G7, Android 10, Firefox 82.0.0-beta.1. I don't know what was the last working version, but it started happening pretty recently (so probably last update broke it, but maybe some update before). I don't see any pattern when it happens, it happens on long websites like news sites or forums.

**Comment by garrett** [comment]
@peter-vasut: Thanks; that helps! On one hand, it shows that I'm not the only one that's affected. And given the timing, it would've hit beta by now.

---

I was only able to reproduce this on my OnePlus 6T running OxygenOS (the default "stock" build of Android that it comes with), and only on Nightly (as of the time of filing this issue — as mentioned in the above comment, it's now on Beta). Turning off all extensions did not improve the issue.

I was not able to reproduce it on my Nexus 6P or Nexus 5, which are both old phones and running 3rd party ROMs. (crDroid & an unofficial LineageOS, both on Android 10).

I was able to reproduce the issue more frequently on my 6T by swiping down from the top of the screen (similar to seeing notifications) in the middle of a scroll. That stopped the scrolling and felt similar to this bug, but is normal across all versions of Firefox on all the phones I could test. On Nightly on my OnePlus 6T, this caused the scrolling bug to show up more frequently afterward. My other devices were unaffected afterward, even on Nightly.

As my OnePlus 6T had seen a lot of usage over the past 2+ years, I figured a factory reset was in order anyway (for other reasons). I went further and installed LineageOS on it over the weekend. It's no longer affected by this issue, apparently, even after using Firefox Sync and adding all my extensions back.

---

So, a summary of what we know so far:

- Some devices have versions of Android where Firefox (Nightly and now Beta) rejects touch scrolling intermittently.
- It's related to a change in the engine that landed in Fenix somewhere around the last week of August.
- It seems to affect some people with some phones, but not all.

What it is not:

- It's not related to an extension.
- It's not related to the first paint secret toggle.
- It's not related to WebRender being on or off.
- It's not related to gesture navigation, as it also happens with the old three-button navigation.
- It's not related to the hardware of a phone, but the software stack running on it.
- It's not related to old, possibly corrupted data (as I removed data and did a re-sync and still had the issue).
- It's not in Firefox stable yet, but _is going to be_, unless it's found and fixed and hotfixed.

My fear: As the buggy code will hit stable within a few weeks — unless stopped — many more people (with affected phones) will probably hit this problem, and many may just switch away from Firefox when they cannot scroll. :scream: 

Were there any scrolling and/or touch related changes that landed in Gecko toward the end of August that might trigger this? (Or perhaps earlier, if Firefox components were sync'd around then.)

**Comment by staktrace** [comment]
Please try using the GeckoViewExample app and see if it reproduces there. The most recent one (as of this writing) can be obtained from https://firefox-ci-tc.services.mozilla.com/api/queue/v1/task/XWO4gTUEQniktUZ9kszI_A/runs/0/artifacts/public/build/geckoview_example.apk

If it does reproduce there, it would be great to use mozregression to find out the exact change that introduced this problem. I can walk you through how to do that if you need.

**Comment by Mugurell** [comment]
@garrett @peter-vasut Any websites in particular where you've seen this happening?

**Comment by peter-vasut** [comment]
> Please try using the GeckoViewExample app and see if it reproduces there.

I tried to browse some websites for few minutes and I couldn't reproduce it. I can't confirm that it doesn't happen for sure, because I don't know circumstances that trigger that. In Firefox beta, it happens randomly, but it happens pretty often, so I guess I would find out by now if same thing happened in GeckoViewExample. (I have tried many sites I frequently visit just in case, but problem doesn't seem to be site dependent.)

**Comment by Mugurell** [comment]
> > That change was in the search/address bar UI. This sounds like Gecko web content interactions.
> 
> Either that or maybe our gesture detector, @Mugurell?

Based on 

> As Nightly has the option to not auto-hide the URL bar, I enabled that and tried scrolling pages. It still happens, even in
> reader mode. So this probably has nothing to do with hiding or showing the URL bar.

I'd rule out the `BrowserGestureDetector` which animates the toolbar (only used for dynamic bottom toolbar) (but could have the potential to cause something like this since it acts upon touches on Main .. and maybe it could block Main.. though it should be very lean..).

And based on 
> When the URL bar is hidden, it brings up the URL bar, but doesn't scroll the page. When the URL bar is constantly shown, this not-scrolling swipe still happens, however.

I'd say it's clear that the app is responding to touches (if the url bar shows up), Main is not blocked but GV doesn't scroll.

**Comment by Mugurell** [comment]
Based on https://github.com/mozilla-mobile/fenix/issues/14527#issuecomment-696856587
> It does happen in Nightly, it does not happen in Beta (yet), and doesn't happen in Release.

Maybe https://github.com/mozilla-mobile/android-components/issues/8209 which uses https://bugzilla.mozilla.org/show_bug.cgi?id=1633322 might be causing this...?
The timing seems to fit perfectly for both Nightly and Beta (beta got this a week ago and so did the reports for beta).
Since [we're forwarding all (other than ACTION_DOWN) MotionEvents to GV ](https://github.com/Mugurell/android-components/commit/2592dc7132dca7e5b870345a4cf1918af06797a3#diff-911d1c7181654a2087d8a3ee402efd7eR107) maybe there's an edgecase in GV that could cause this? @snorp

**Comment by Mugurell** [comment]
Thank you @garrett @peter-vasut @TheLounger for the details here. They really help.
If you have any more details about the situations in which this issue might occur please continue posting.

Also tagging @sflorean to raise awareness about this, maybe in day to day testing QA is also experiencing this. 
I'd be mostly interested in websites on which this reproduces, position in page (top, middle, bottom), elements over which the scroll happens (text, image, iframe), toolbar state (top/bottom, fixed/dynamic, whether it is animated while the page does not scroll), anything that could help in pinpointing a particular scenario for this.

**Comment by peter-vasut** [comment]
@Mugurell 
* affected websites: [Here](https://sport.aktuality.sk/c/462677/sportovemu-riaditelovi-slovenskej-reprezentacie-na-ms-v-imole-bolo-po-pretekoch-muzov-takmer-do-placu/?AT=wgt.hp_hp-najcitanejsie-3hod.c.x...B.&utm_source=aktuality.sk&utm_medium=zona-hp&utm_content=box-top-hp-najcitanejsie&utm_campaign=cross) is random link to Slovak news site that causes problem. Even visiting plain html-only page with basic elements like text, table and images is problematic (not publicly accessible so I can't share). For some reason, I can't reproduce problem on Reddit homepage, but pages with individual posts cause problems. Most websites are problematic, Reddit is odd one, I have noticed it only recently (i'll update if Reddit stops working).
* bug affects scrolling, zooming and probably also clicking links
* problem seems to happen on any position on page over any element (tested text, images, also in reader mode)
* affects both portrait and landscape orientation, changing screen resolution doesn't help either
* doesn't matter if scrolling up or down, I have tried to disable toolbar hiding and put it on top, doesn't help
* If status bar is set to automatically hide, when I cannot scroll, statusbar unhides immediately after touching screen, even if I try to scroll up. If it's already visible, it stays visible.
* I have tried to change system language to English, disable Gboard keyboard, I don't know what else could be different on my device from most people. Nothing helps.

**Comment by garrett** [comment]
> Even visiting plain html-only page with basic elements like text, table and images is problematic

When I had the problem on my phone, on the stock OS (OxygenOS, based on Android 10), I hit the issue across pretty much all websites, in reader mode, and even when panning around an image (when viewing just an image itself; no HTML or anything else).

> If status bar is set to automatically hide, when I cannot scroll, statusbar unhides immediately after touching screen, even if I try to scroll up. If it's already visible, it stays visible.

I noticed that too. And setting the URL bar to _not_ auto-hide still didn't work around the problem.

>  I don't know what else could be different on my device from most people. Nothing helps.

I had the same experience. Switching from OxygenOS (the flavor of Android OnePlus ships) to LineageOS (AOSP + some additions) fixed it to me. We can't expect anyone else to do this, naturally. :wink: But it shows that the problem isn't triggered by the hardware and probably not even in the drivers, as LineageOS has to use the same blobs the vendors ship to get hardware support working.

Therefore, there's probably something in the customization that various vendors ship that happen to trigger this issue in Firefox — and only Firefox. Or it's something in Android that Firefox doesn't play well with, which happens to hit certain users of Firefox under some unknown circumstances.

Note: I never saw this problem in any other app, including Fennec or previous versions of Fenix either. So it's definitely a Firefox bug.

A wild guess: I did notice that OxygenOS gesture support worked even when apps would fullscreen themselves (such as games), but gestures on LineageOS (+ GApps) do not do this; they require a first swipe to get out of fullscreen and then a second for the gesture. Google did something weird with gesture support so it's tied into launchers _and_ with a custom override that they — and other vendors — ship. Perhaps input handling has changed on some devices to accommodate gesture support (and is active even when gestures aren't turned on), and the OnePlus and LG stock ROMs (among others) have this baked in? And Firefox was tested against a phone that didn't?

Does anyone within Mozilla have a LG or OnePlus phone with Android 10 on the stock OS who could test this out?

**Comment by peter-vasut** [comment]
Do we know in which version this started to happen? Is there some place where I could download daily builds of Firefox for Android? I found some mozilla ftp server with daily builds for desktop Firefox, is there something similar for Android? (I can install Firefox Nightly without affecting my beta install and test which versions are problematic.)

**Comment by solomonfentonmiller** [comment]
I just got the 82.1.1 (Build #2015770923) update on Firefox Daylight on my LG V50 with Android 10 and I started to notice this issue! It doesn't seem to be consistent. Sometimes it's quite bad and scrolling freezes every 3-4 new scrolls and other times there's no problem for a while. I could make a video if that would help.

**Comment by solomonfentonmiller** [comment]
I made a video: https://streamable.com/wrudpq
As you can see, this breaks the experience of using Firefox.

**Comment by yoasif** [comment]
Devices listed [here](https://bugzilla.mozilla.org/show_bug.cgi?id=1419053#c51) include:

* LG G8
* LG V60
* LG G7

**Comment by peter-vasut** [comment]
It's happening in release. I got update to 82.1.1. Such serious bug shouldn't get into release, this makes problematic basically any interaction with browser including scrolling, clicking links, swiping (for example DuckDuckGo image search results), ...

I do understand this is probably some weird race-condition or something, but could it get higher priority? If I understand correctly, we don't even know which exact build introduced the error. If someone can build older Nightly versions (if there is no archive), I can test them (if it doesn't break my regular install). Even if I tested one version per day, we could probably find problematic commit in less than a week.

**Comment by peter-vasut** [comment]
> Devices listed [here](https://bugzilla.mozilla.org/show_bug.cgi?id=1419053#c51) include:
> 
>     * LG G8
> 
>     * LG V60
> 
>     * LG G7

@yoasif You probably provided wrong link, there is no mention of LG phones.

**Comment by jsilverstreak** [comment]
> 
> 
> Do we know in which version this started to happen? Is there some place where I could download daily builds of Firefox for Android? I found some mozilla ftp server with daily builds for desktop Firefox, is there something similar for Android? (I could install Firefox Nightly without affecting my beta install and test which versions are problematic.)

Nightly 200829 06:05 is the first one based on 82 and the first one i could find that has the problem. Nightly 200828 does not have the problem.
LG G8 Android 10.

**Comment by ptar** [comment]
I got a LG V40 Android 10 with Firefox 82.1.1 (Build #2015770923, most current version from play store) and this bug renders Firefox almost useless when reading sites or articles: You just don't know, when tipping on a link or scrolling or zooming in a map works or not.
Really frustrating...

**Comment by hemynem** [comment]
I have a LG G8s with Android 10 same problems on my device. 82.1.1 (Build #2015770923)

**Comment by 1r00t** [comment]
I have it too. Stock Android 10 on Sony Experia 1. So not just LG phones.

**Comment by TheLounger** [comment]
Just noting here that I also have a Sony Xperia 1.

**Comment by Mugurell** [comment]
Hello,
Thank you for the reports, we're still having trouble reproducing this, seems to only be happening sporadically.
If this is an issue that is easily reproduced by you can you please also try on [GVE](https://firefox-ci-tc.services.mozilla.com/tasks/index/gecko.v2.mozilla-central.latest.mobile/android-aarch64-gcp-opt)(geckoview-example.apk) and report back if you see it also happening there?
That is a stripped down version that should eliminate some possible reasons for this.

Also I was seeing reports that this is happening with the toolbar placed at the top or at the bottom. This is an important detail that would be nice to be confirmed.
As always, videos that would show this happening / websites on which this issue is more prevalent are welcomed

**Comment by 1r00t** [comment]
> Hello,
> Thank you for the reports, we're still having trouble reproducing this, seems to only be happening sporadically.
> If this is an issue that is easily reproduced by you can you please also try on [GVE](https://firefox-ci-tc.services.mozilla.com/tasks/index/gecko.v2.mozilla-central.latest.mobile/android-aarch64-gcp-opt)(geckoview-example.apk) and report back if you see it also happening there?
> That is a stripped down version that should eliminate some possible reasons for this.
> 
> Also I was seeing reports that this is happening with the toolbar placed at the top or at the bottom. This is an important detail that would be nice to be confirmed.
> As always, videos that would show this happening / websites on which this issue is more prevalent are welcomed

Hi. I installed that GVE apk. I don't have that problem there. I too think it has something to do with the toolbar.

**Comment by jsilverstreak** [comment]
I installed GVE as well and couldn't reproduce the problem. 
Using Firefox Nightly 201031 17:02 I can easily reproduce it on https://github.com/mozilla-mobile/fenix/issues/14527 and https://en.m.wikipedia.org/wiki/Main_Page but not https://github.com/mozilla-mobile/fenix/issues/9773.
Also It happens with both top and bottom toolbar. With the top toolbar the animation blocks the website as if bottom toolbar was also there before snapping to correct not sure if related. [Gif](https://thumbs.gfycat.com/HeartfeltAntiqueBuckeyebutterfly-size_restricted.gif)

**Comment by solomonfentonmiller** [comment]
I also installed GVE and the issue does not occur there!

I can confirm on 82.1.1 that the bug occurs with the toolbar on top, bottom and with scroll to hide or not. No difference with any of those settings. As you can see in the video I posted earlier, the scrolling freeze always causes the toolbar to show, even if it shouldn't.

Tell me if I can help in other ways. I'm afraid that users with this issue are already moving to Chrome.

**Comment by garrett** [comment]
> Also I was seeing reports that this is happening with the toolbar placed at the top or at the bottom. This is an important detail that would be nice to be confirmed.

@Mugurell: When I was having this problem on my phone (before I switched to LineageOS on my OnePlus 6T and Firefox stopped having issues), it also happened with the toolbar on top. It did _feel_ like it happened with gesture navigation and toolbar at the bottom _more_, but it definitely happened with 3-button classic navigation and the toolbar at the top too... so it could have been perception, random luck, or perhaps one or both of these things might influence it. Or not at all? :man_shrugging:

**Comment by piranhaphish** [comment]
Same issue on OnePlus 6T, Android 10, FF 82.1.3.

It's not just scrolling, which I am experiencing as well. Punch/zoom gestures respond only part of the time. Using swipe typing on the keyboard is ignored sometimes. Even clicks/taps seem to get missed.

I cleared the data for FF and that seemed to help for a few hours, but it occurred again. I know that's not really empirical.

I have to say, this makes FF maddening to use. I honestly can't handle it and will reluctantly be switching to another browser for the time being.

**Comment by snoukkis** [comment]
Same issue on OnePlus 6, Android 10, FF 82.1.3.

**Comment by henrypa** [comment]
Sorry for the non-informatical post, but I can confim this problem on OnePlus 6T, latest Android 10 and the latest Firefox 82.1.3 android (stable) version. Pinch zooming and scrolling is near impossible at this moment, tried pretty much everything from changing keyboard software, disabling oneplus gestures, etc.. doesn't seem to matter. It is so bad that I moved to the Brave browser.

I don't have the problem in geckoview-example.apk. This bug started sometime ago already, I believe jsilverstreak's findings are in line with me:  https://github.com/mozilla-mobile/fenix/issues/14527#issuecomment-719998932

**Comment by vecchionico** [comment]
I had to revert to 81.1.5 to avoid this bug. OnePlus 6, Android 10, Oxygen 10.3.6

**Comment by m-a-orlowski** [comment]
I have the same problem on LG G7 ThinQ with Android 10 (though it probably happened also on Android 9, before I updated). The geckoview-example also works fine for me. I don't think the placement of the toolbar matters here.
I hope this gets fixed soon as it's currently the only thing that keeps me staying on Firefox 68.

**Comment by ptar** [comment]
> [I got a LG V40 Android 10 with Firefox 82.1.1...](https://github.com/mozilla-mobile/fenix/issues/14527#issuecomment-720024957)

I checked out the GVE(geckoview-example.apk) today and can't reproduce this issue with it. But it seems to be gone away with FF 82.1.3, too, which I wanted to use in comparison.

I know, I had trouble with this issue during and shortly after the US election on the German news portal https://www.spiegel.de/thema/us_praesidentschaftswahl_2020/

Currently, I can't reproduce it anymore on that site.

I will now make FF the default browser (again), clear the cache and try if it re-occurs on other sites I use frequently.

If you fixed it and read this: Thank you very much! 👍

**Comment by ptar** [comment]
Mhmm, that was quick: Obviously, the problem occurs less often, but it still occurs.
It happened with the article on https://www.heise.de/news/Volkswagen-investiert-Milliarden-in-E-Mobilitaet-und-Digitales-4960473.html, where the issue occurs. Not as often as before, but it definitely occurs. It does not with the GVE(geckoview-example.apk).
I hope you can track down the bug!

**Comment by snoukkis** [comment]
As mentioned above, this is unfortunately very much happening on 82.1.3. I have experienced it constantly today on most sites I've visited.
Disabling address bar hiding makes it slightly tolerable: instead of the page jerking, it simply misses a touch. It's what's bearly keeping me from switching browsers.

**Comment by sskodje** [comment]
It happens all the time on 82.1.3 on LG G7. 5-6 seconds between perhaps. Disabling address bar hiding doesn't really do anything to fix the problem, but it removes the animation at least. It's still a crapshoot if clicking links or scrolling works.

**Comment by Dan-Wun** [comment]
On Xiaomi Mi Mix 2s, Android 8.0.0, it happens permanently, every third, fourth try to scroll. Same with beta and nightly, browsing is like impossible. 
Problem first occured some weeks ago, no Problem with Gecko view example.
I will sadly have to switch to another Browser. Or downgrade to the last working version temporalily. Is it known, which version that is? I really hope, this gets fixed...

**Comment by peter-vasut** [comment]
Apparently, Mozilla has some [CI](https://firefox-ci-tc.services.mozilla.com/tasks/index/mobile.v2.fenix.debug) system that allows to download any build. I have downloaded and tested various versions to find first bad one. Thanks to @jsilverstreak for narrowing down the day.

**Last working version**: [ff805e53eda2594bdb311d4bb8dd38e40cc7a219](https://firefox-ci-tc.services.mozilla.com/tasks/index/mobile.v2.fenix.debug.2020.08.28.revision.ff805e53eda2594bdb311d4bb8dd38e40cc7a219) - tested for few hours
**First bad version**: [487eeb6b88007766e77872cb28c6f819b242f378](https://firefox-ci-tc.services.mozilla.com/tasks/index/mobile.v2.fenix.debug.2020.08.29.revision.487eeb6b88007766e77872cb28c6f819b242f378)

If anyone confirms these findings, add :+1: or something to this post, maybe it's just coincidence that ff805e5 works for me (even though it's highly unlikely it works for hours by chance). It's weird, because the first bad commit looks like only changing some strings, so idk.

**Comment by sskodje** [comment]
> I will sadly have to switch to another Browser. Or downgrade to the last working version temporalily. Is it known, which version that is? I really hope, this gets fixed...

I installed 81.1.5 from apkmirror.com and it works fine. I think that is the last working release.

**Comment by anamba** [comment]
A couple additional data points, I have been experiencing this issue on my LG G7+ and LG V50, both running the latest software available (Android 10). I tried the `ff805e5` build mentioned above on my V50 and the issue is _not_ present in that version.

**Comment by jsilverstreak** [comment]
9620763 appears to be the commit that causes this problem, in case this wasn't already known.

**Comment by Mugurell** [comment]
Thank you for all your help and support here!

Based on your reports I found a scenario which would show scrolling not working on a OnePlus7T but interestingly that same scenario shows no issue on a Pixel3.
Posted a patch that would fix this and I'm confident it would also fix the other scenarios you are seeing.
The fix should be available in the next Nightly.

**Comment by anamba** [comment]
I tested this version: https://firefox-ci-tc.services.mozilla.com/tasks/index/mobile.v2.fenix.debug.2020.11.16.revision.1d5f74d865250f05253f86f73d6bfd0c33e1f1aa/arm64-v8a

Same issue... scrolling down after a pause of several seconds is not reliable (shows toolbar) and tapping links is not reliable either (usually requires a 2nd tap).

**Comment by Mugurell** [comment]
Thanks anamba.
Indeed it seems the AC change is not yet part of the current Nightly (November 17th).
Should be in the next one - November 18th.

**Comment by anamba** [comment]
It works!! https://firefox-ci-tc.services.mozilla.com/tasks/index/mobile.v2.fenix.nightly.2020.11.17.latest/arm64-v8a

**Comment by AndiLeni** [comment]
@Mugurell 
The issue seems to be fixed in the newest nightly release. 

What caused the problem?
Can you link the git commit that solved it?

**Comment by Mugurell** [comment]
> What caused the problem?
> Can you link the git commit that solved it?

This was hopefully resolved upstream, in AC by these changes - https://github.com/mozilla-mobile/android-components/pull/8988/files
Like in the case of all cool bugs the issue was not really obvious. Seems like a race condition fixed by using callbacks instead of coroutines for interacting with GeckoView following users touching the screen.

**Comment by Dan-Wun** [comment]
Great, that @Mugurell could fix that so quickly.
It took me weeks to find this thread, I thought my phone was the culprit.
As I am on 81.1.1.5, as @sskodje suggested (thanks!): do I have to check release notes to know when the main version is fixed?

**Comment by Mugurell** [comment]
> Great, that @Mugurell could fix that so quickly.
> It took me weeks to find this thread, I thought my phone was the culprit.
> As I am on 81.1.1.5, as @sskodje suggested (thanks!): do I have to check release notes to know when the main version is fixed?

Don't think it'll make it into the release notes.
But should be included in all next releases: Release(83), Beta(84) and Nightly(85).

**Comment by henrypa** [comment]
Thanks for the quick fix @Mugurell ! Seems to work just fine on Oneplus 6T.

**Comment by AndiLeni** [comment]
@Mugurell 
I installed version 83.1.0 from github, but unfortunately the problem still seems to exist :(

Is the fix shipped in a future version?

**Comment by mm1ck** [comment]
FF just updated from the play store to 83.1.0. 
As above the scroll bug still exists.
Sony Xperia 1, android 10.

Has the fix been rolled out, or are we still waiting on more testing.

**Comment by 1r00t** [comment]
> FF just updated from the play store to 83.1.0.
> As above the scroll bug still exists.
> Sony Xperia 1, android 10.
> 
> Has the fix been rolled out, or are we still waiting on more testing.

Can confirm. Still present on Xperia 1 and FF 83.1.0.

**Comment by Mugurell** [comment]
The fix talked about did not make it to `Firefox 83.1.0`
but it is present in `Firefox Beta 84.0.0-beta.1`
And the current Nightlies.

**Comment by mm1ck** [comment]
> The fix talked about did not make it to `Firefox 83.1.0`
> but it is present in `Firefox Beta 84.0.0-beta.1`
> And the current Nightlies.

Can confirm that the issue is gone with the latest nightly on the play store.

Sony Xperia 1, android 10.

Nice to see the tabs grid view is back, but shame that you can't move them around like we used to.

**Comment by sflorean** [comment]
Verified as fixed on Nightly 11/23 with:
- LG G7 (Android 8.0)
- Pixel 3 (Android 11)
